src/components/points-prelevement/volumes-chart.js

```diff
@@ -0,0 +1,128 @@
+'use client'
+
+import {useState} from 'react'
+
+import {
+  LineChart,
+  ChartsReferenceLine
+} from '@mui/x-charts'
+import {format, parseISO} from 'date-fns'
+import {fr} from 'date-fns/locale'
+
+const VolumesChart = ({volumes}) => {
+  const [showAll, setShowAll] = useState(false)
+  const sortedData = [...volumes.valeurs].sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime())
+  const displayData = showAll ? sortedData : sortedData.slice(-12)
+  const xLabels = displayData.map(item => item.date)
+  const volumeData = displayData.map(item => item.volume)
+  const exceededData = displayData.map(item => ({
+    date: item.date,
+    volume: item.volume > volumes.volumeJournalierMax ? item.volume : null
+  }))
```

Comment:
Est-ce qu'il ne vaudrait pas mieux filtrer directement `displayData` ?
```suggestion
const exceededData = displayData
  .filter(item => item.volume > volumes.volumeJournalierMax)
  .map(item => ({
    date: item.date,
    volume: item.volume
  }));
```

Reply:
Apparemment, ça ne fonctionne pas (tous les points sont en bleu).
Je crois qu'on ne peut pas filtrer dans les series directement.